File: backend/internal/models/post.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/datatypes"
	"gorm.io/gorm"
)

type PostType string

const (
	PostTypeArticle PostType = "article"
	PostTypePoll    PostType = "poll"
	PostTypeVideo   PostType = "video"
	PostTypeEmbed   PostType = "embed"
	PostTypeLink    PostType = "link"
	PostTypeGallery PostType = "gallery"
)

type Post struct {
	ID           uuid.UUID      `gorm:"type:uuid;primaryKey" json:"id"`
	UserID       uuid.UUID      `gorm:"type:uuid;not null" json:"user_id"`
	User         User           `gorm:"foreignKey:UserID" json:"user,omitempty"`
	CategoryID   uint           `gorm:"not null" json:"category_id"`
	Category     Category       `gorm:"foreignKey:CategoryID" json:"category,omitempty"`
	Title        string         `gorm:"not null;size:255" json:"title"`
	Body         string         `gorm:"type:text" json:"body"`
	PostType     PostType       `gorm:"type:varchar(20);default:'article'" json:"post_type"`
	Metadata     datatypes.JSON `gorm:"type:jsonb" json:"metadata,omitempty"`
	LikeCount    int            `gorm:"default:0"     json:"like_count"`
	CommentCount int            `gorm:"default:0"     json:"comment_count"`
	Pinned       bool           `gorm:"default:false" json:"pinned"`
	PinnedAt     *time.Time     `                     json:"pinned_at,omitempty"`
	CreatedAt    time.Time      `json:"created_at"`
	UpdatedAt    time.Time      `json:"updated_at"`
	DeletedAt    gorm.DeletedAt `gorm:"index" json:"-"`
}

File: mobile/src/types/post.ts
import type {User} from './user';

export type PostType = 'article' | 'poll' | 'video' | 'embed' | 'link' | 'gallery';

export interface Post {
  id: string;
  user_id: string;
  user: User;
  category_id: number;
  title: string;
  body: string;
  post_type: PostType;
  metadata: Record<string, unknown> | null;
  like_count: number;
  comment_count: number;
  pinned: boolean;
  created_at: string;
  updated_at: string;
}

export interface Comment {
  id: string;
  post_id: string;
  user_id: string;
  user: User;
  parent_id: string | null;
  body: string;
  like_count: number;
  created_at: string;
}
